packages/edit-site/src/components/sidebar/spacing-panel.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	__experimentalBoxControl as BoxControl,
+	PanelBody,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { useEditorFeature } from '../editor/utils';
+
+export function useHasSpacingPanel( { supports } ) {
+	return supports.includes( 'paddingBottom' );
+}
+
+function filterUnitsWithSettings( settings = [], units = [] ) {
+	return units.filter( ( unit ) => {
+		return settings.includes( unit.value );
+	} );
+}
+
+function useCustomUnits( units ) {
+	const availableUnits = useEditorFeature( 'spacing.units' );
+	const usedUnits = filterUnitsWithSettings(
+		! availableUnits ? [] : availableUnits,
+		units
+	);
+
+	return usedUnits.length === 0 ? false : usedUnits;
+}
+
+export default function SpacingPanel( {
+	context: { name },
+	getStyleProperty,
+	setStyleProperty,
+} ) {
+	const units = useCustomUnits();
+	const paddingValues = {
+		top: getStyleProperty( name, 'paddingTop' ),
+		right: getStyleProperty( name, 'paddingRight' ),
+		bottom: getStyleProperty( name, 'paddingBottom' ),
+		left: getStyleProperty( name, 'paddingLeft' ),
+	};
+	const setPaddingValues = ( { top, right, bottom, left } ) => {
```

Comment:
When the user is using the default BoxControl state that changes all the padding values at the same time, there is only a single user action changing all the padding values but here we are making four setStyleProperty calls. The fact that we call setStyleProperty multiple times when the user just made one change makes undo/redo not work as expected.

E.g: Set padding to 30 (all padding values) than set to 50. Press undo instead of undoing all the padding values to 30 the undo just affects paddingLeft. At the block level, it seems this issue is not happening, and undo changes all the padding values.